lib/compiler/src/beam_ssa_type.erl

```diff
@@ -1365,6 +1385,21 @@ will_succeed_1(#b_set{op=get_tuple_element}, _Src, _Ts) ->
     yes;
 will_succeed_1(#b_set{op=put_tuple}, _Src, _Ts) ->
     yes;
+will_succeed_1(#b_set{op=update_tuple,args=[Tuple | Updates]}, _Src, Ts) ->
+    case beam_types:meet(normalized_type(Tuple, Ts), #t_tuple{}) of
+        none ->
+            no;
```

Comment:
Not covered by the compiler test suite. Lines 1387 and 1395 are also not covered.